### PR TITLE
feat(construct): new construct for deployment of custom models to sagemaker

### LINKED ISSUE #150: (New construct): Add support to deploy custom models to a sagemaker endpoint
### Describe the feature

Add a construct to enable deployment of custom models to a SageMaker endpoint (fine tuned, inf2,...)
This construct should provide the user the ability to point towards a location in Amazon S3 where the model artifacts are located, and a DLC container image

### Use Case

More flexibility and customization for end users

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### LINKED ISSUE #179: (gen-ai): JumpStartSageMakerEndpoint not working
### Describe the bug

I'm unable to deploy the SageMaker endpoint using the `JumpStartSageMakerEndpoint` construct.

### Expected Behavior

The endpoint gets created.

### Current Behavior

CloudFormation throws the following error when trying to create the endpoint:
```
Could not find model data at s3://jumpstart-cache-prod-us-east-1/huggingface-asr/huggingface-asr-whisper-base/artifacts/inference-prepack/v1.0.1/. (Service: AmazonSageMaker; Status Code: 400; Error Code: ValidationException; Request ID: 8e830934-e61a-4c1e-96e5-1663211204b0; Proxy: null)
```

error

### Reproduction Steps

```typescript
import * as cdk from "aws-cdk-lib";
import { Construct } from "constructs";
import {
  JumpStartModel,
  SageMakerInstanceType,
  JumpStartSageMakerEndpoint,
} from "@cdklabs/generative-ai-cdk-constructs";

export class SagemakerJumpStartFmEndpointStack extends cdk.Stack {
  jumpStartEndpoint: JumpStartSageMakerEndpoint;

  constructor(scope: Construct, id: string, props: SagemakerJumpStartFmEndpointStackProps) {
    super(scope, id, props);

    this.jumpStartEndpoint = new JumpStartSageMakerEndpoint(this, "JumpStartEndpoint", {
      model: JumpStartModel.HUGGINGFACE_ASR_WHISPER_BASE_2_0_0,
      instanceType: SageMakerInstanceType.of("ml.g5.2xlarge"),
    });
}
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### CDK CLI Version

2.111.0

### Framework Version

0.1.11

### Node.js Version

21.4.0

### OS

Mac OS

### Language

Typescript

### Language Version

_No response_

### Region experiencing the issue

us-east-1

### Code modification

No

### Other information

_No response_

### Service quota

- [X] I have reviewed the service quotas for this construct

### PR DESCRIPTION
**Pull request by krokoko**
First version of a construct allowing to deploy models stored in a custom S3 location to an Amazon SageMaker endpoint. 
- New construct CustomSagemakerEndpoint Fixes #150 
- Update documentation 
- Add new projen task to run the code generation step (create list of available containers / jumpstart models). Can be used through ```projen generate-models-containers```
- Update list of available models
- Fixes #179 : address the issue of some models from JumpStart which can't be deployed. This is due to the construct requiring model data to be provided as a compressed object, however some newer JumpStart models provide uncompressed model artifacts. Tested the fix with the reported model - whisper base 2.0.0. Deployed model and ran inference. This requires an upgrade of cdk as the update to the L1 CloudFormation resource was added recently (addition of the property https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_sagemaker.CfnModel.ModelDataSourceProperty.html) 

Once this is merged, I will add samples to the dedicated repo

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

**Review by dineshSajwan — COMMENTED** [review]
Over all looks good , added some small feedback comments.

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/README_custom_sagemaker_endpoint.md [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    instanceCount: 1,
+    volumeSizeInGb: 100
+  });
+```
+
+## Initializer
+
+```
+new HuggingFaceSageMakerEndpoint(scope: Construct, id: string, props: IHuggingFaceSageMakerEndpointProps)
+```
```
Comment (dineshSajwan) [inline src/patterns/gen-ai/aws-model-deployment-sagemaker/README_custom_sagemaker_endpoint.md:71]: shouldn't this be CustomSageMakerEndpoint in place of HuggingFaceSageMakerEndpoint ?

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/README_hugging_face.md [hunk 2]
```diff
@@ -85,30 +77,8 @@ Parameters
 - id string
 - props IHuggingFaceSageMakerEndpointProps
```
Comment (dineshSajwan) [inline src/patterns/gen-ai/aws-model-deployment-sagemaker/README_hugging_face.md:79]: HuggingFaceSageMakerEndpoint initializer is missing.

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/README_hugging_face.md [hunk 3]
```diff
@@ -124,20 +94,6 @@ Parameters
```
Comment (dineshSajwan) [inline src/patterns/gen-ai/aws-model-deployment-sagemaker/README_hugging_face.md:94]: are you planning to update the props later?

**Review by dineshSajwan — COMMENTED** [review]
Nit picks.

File: .projenrc.ts [hunk 4]
```diff
@@ -27,7 +27,7 @@ import {
 const GITHUB_USER = 'awslabs';
 const PUBLICATION_NAMESPACE = 'cdklabs';
 const PROJECT_NAME = 'generative-ai-cdk-constructs';
-const CDK_VERSION: string = '2.88.0';
+const CDK_VERSION: string = '2.118.0';
```
[the remark this answers is not in the document]
Comment (krokoko) [reply .projenrc.ts:30]: Yes, I am using a property which was introduced recently (2.117 if I'm correct): https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_sagemaker.CfnModel.ModelDataSourceProperty.html

File: .projen/deps.json [hunk 5]
```diff
@@ -87,7 +87,7 @@
     },
     {
       "name": "aws-cdk-lib",
-      "version": "^2.88.0",
+      "version": "^2.118.0",
```
Comment (scottschreckengaust) [inline .projen/deps.json:90]: Is this the lowest version of the CDK required?

Comment (scottschreckengaust) [reply .projen/deps.json:90]: I found that the solution does not compile below `2.114.0`

Comment (scottschreckengaust) [inline .projen/deps.json:90]:
```suggestion
      "version": "^2.114.0",
```

File: .projenrc.ts (hunk 4 above)
Comment (scottschreckengaust) [inline .projenrc.ts:30]:
```suggestion
const CDK_VERSION: string = '2.114.0';
```

File: .projenrc.ts [hunk 6]
```diff
@@ -151,5 +151,17 @@ project.eslint?.addRules({
   'header/header': [2, 'header.js'],
 });
 
+// Add generation of new available models for constructs
+project.addTask('generate-models-containers', {
+  description: 'Generate new list of models available from Jumpstart and DLC containers',
+  steps: [
+    {
+      say: 'Generate new list of models available from Jumpstart and DLC containers',
+      cwd: project.srcdir+'/patterns/gen-ai/aws-model-deployment-sagemaker/code-generation',
```
[the remark this answers is not in the document]
Comment (krokoko) [reply .projenrc.ts:160]: Yes that folder is already part of the code base, just adding a task is projen to simplify the call to that

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,7 @@
+# CDK Generative AI Constructs V0.1.20 (2024-01-10)
+
+Based on CDK library version 2.118.0
```
Comment (scottschreckengaust) [inline CHANGELOG.md:3]:
```suggestion
Based on CDK library version 2.114.0
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,3 +1,7 @@
+# CDK Generative AI Constructs V0.1.20 (2024-01-10)
```
Comment (scottschreckengaust) [inline CHANGELOG.md:1]:
```suggestion
# CDK Generative AI Constructs V0.1.20 (2024-01-11)
```

Comment (krokoko) [reply CHANGELOG.md]: done in [e8ebbf5](https://github.com/awslabs/generative-ai-cdk-constructs/pull/180/commits/e8ebbf58d89f9699465624722f45c3eb9401228f)

File: CHANGELOG.md (hunk 7 above)
Comment (krokoko) [reply CHANGELOG.md]: done in [e8ebbf5](https://github.com/awslabs/generative-ai-cdk-constructs/pull/180/commits/e8ebbf58d89f9699465624722f45c3eb9401228f)

File: DEVELOPER_GUIDE.md [hunk 9]
```diff
@@ -16,9 +16,9 @@ Default output format [None]: json
 ```
 
 - [Node](https://nodejs.org/en) >= v20.9.0
-- [AWS CDK](https://github.com/aws/aws-cdk/releases/tag/v2.102.0) >= 2.111.0
+- [AWS CDK](https://github.com/aws/aws-cdk/releases/tag/v2.102.0) >= 2.118.0
```
[the remark this answers is not in the document]
Comment (krokoko) [reply DEVELOPER_GUIDE.md]: done in [e8ebbf5](https://github.com/awslabs/generative-ai-cdk-constructs/pull/180/commits/e8ebbf58d89f9699465624722f45c3eb9401228f)

File: .projenrc.ts (hunk 4 above)
Comment (krokoko) [reply .projenrc.ts]: done in [e8ebbf5](https://github.com/awslabs/generative-ai-cdk-constructs/pull/180/commits/e8ebbf58d89f9699465624722f45c3eb9401228f)

File: .projen/deps.json (hunk 5 above)
Comment (krokoko) [reply .projen/deps.json]: done in [e8ebbf5](https://github.com/awslabs/generative-ai-cdk-constructs/pull/180/commits/e8ebbf58d89f9699465624722f45c3eb9401228f)

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/README_hugging_face.md (hunk 2 above)
Comment (krokoko) [reply src/patterns/gen-ai/aws-model-deployment-sagemaker/README_hugging_face.md:79]: done in [e8ebbf5](https://github.com/awslabs/generative-ai-cdk-constructs/pull/180/commits/e8ebbf58d89f9699465624722f45c3eb9401228f)

File: src/patterns/gen-ai/aws-model-deployment-sagemaker/README_custom_sagemaker_endpoint.md (hunk 1 above)
Comment (krokoko) [reply src/patterns/gen-ai/aws-model-deployment-sagemaker/README_custom_sagemaker_endpoint.md]: done in [e8ebbf5](https://github.com/awslabs/generative-ai-cdk-constructs/pull/180/commits/e8ebbf58d89f9699465624722f45c3eb9401228f)

**Review by dineshSajwan — APPROVED** [review]
LGTM

**Review by scottschreckengaust — APPROVED** [review]

**Pull request merged**